src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -109,22 +112,205 @@ export default class NativeAdapter implements IMediaSourceAdapter {
   }
 
   /**
+   * @constructor
+   * @param {IEngine} engine - The video element which bind to NativeAdapter
+   * @param {string} source - The source URL
+   * @param {Object} config - The media source adapter configuration
+   */
+  constructor(engine: IEngine, source: Object, config: Object) {
+    this._engine = engine;
+    this._config = config;
+    this._videoElement = engine.getVideoElement();
+    this._source = source.url;
+    this._eventManager = new EventManager();
+  }
+
+  /**
    * Load the video source
    * @function load
-   * @returns {void}
+   * @returns {Promise<Object>} - The loaded data
    */
-  load(): void {
-    NativeAdapter._logger.debug('load');
-    this._videoElement.load();
+  load(): Promise<Object> {
+    if (!this._loadPromise) {
+      this._loadPromise = new Promise((resolve, reject) => {
+        this._eventManager.listen(this._videoElement, PlayerEvents['LOADED_METADATA'], () => {
```

Comment:
It equals, but I think it better to use `PlayerEvents.LOADED_METADATA` syntax for consistency,